### PR TITLE
Check res middleware

### PR DESCRIPTION
**Pull request by stevkim**
Please look at https://github.com/pokemon-parKing/parKing-reservation-server/pull/11 first. As this branch had the changes from that branch which werent merged yet when making this PR.

The middleware check if we already have a reservation tied to the user's car.
-- One car, one reservation. 

-- For future, this middleware can be changed to check if the reservation exists for a specific time instead.

### CODE REVIEW

**Comment by stevkim** [comment]
Included is the bug-fix that was related to the server only being able to handle one put request at a time.

File: server/middleware/checkReservation.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const { Reservations } = require('../db.js');
+
+const checkReservation = async (req, res, next) => {
```
Comment (clee4037) [inline server/middleware/checkReservation.js:3]: You're a detective! Nice work identifying and handling this case.

File: server/controllers/reservationController.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ const getNearestGarages = async (params) => {
   const { lat, lng } = params;
 
   try {
-    const { data, error } = await Garages
+    const { data, error } = await supabase.from('garages')
```
Comment (clee4037) [inline server/controllers/reservationController.js:12]: Thanks @bzw24 for debugging!

**Review by clee4037 — APPROVED** [review]

**Review by bzw24 — APPROVED** [review]
Good with the changes, we talked this change through at our meeting today, so no comments needed.

**Pull request merged**